BungeeCord-Patches/0049-Add-some-things-to-the-API.patch

```diff
@@ -0,0 +1,264 @@
+From fa435a332e99ec02e25ffb65d90c78aa10bbd172 Mon Sep 17 00:00:00 2001
+From: MrIvanPlays <pekov.ivan@abv.bg>
+Date: Mon, 11 Mar 2019 16:00:17 +0200
+Subject: [PATCH] Add some things to the API
+
+
+diff --git a/api/src/main/java/net/md_5/bungee/api/Callback.java b/api/src/main/java/net/md_5/bungee/api/Callback.java
+index 0cccc175..f383279b 100644
+--- a/api/src/main/java/net/md_5/bungee/api/Callback.java
++++ b/api/src/main/java/net/md_5/bungee/api/Callback.java
+@@ -6,6 +6,7 @@ package net.md_5.bungee.api;
+  *
+  * @param <V> the type of result
+  */
++@FunctionalInterface // Waterfall
+ public interface Callback<V>
+ {
+ 
+diff --git a/api/src/main/java/net/md_5/bungee/api/CommandSender.java b/api/src/main/java/net/md_5/bungee/api/CommandSender.java
+index a35b3fd0..f83fc023 100644
+--- a/api/src/main/java/net/md_5/bungee/api/CommandSender.java
++++ b/api/src/main/java/net/md_5/bungee/api/CommandSender.java
+@@ -16,19 +16,19 @@ public interface CommandSender
+ 
+     /**
+      * Send a message to this sender.
++     * Waterfall - Deprecation removal. No reason deprecating
+      *
+      * @param message the message to send
+      */
+-    @Deprecated
+     public void sendMessage(String message);
```

Comment:
This method is deprecated for valid reasons. Legacy texts are converted to text components, but you should always use the latter.

Reply:
Then why md_5 doesn't specify it as '@deprecated' ?!

Reply:
You can already see it's deprecated from the annotation on the method; clearly he didn't feel it was necessary to duplicate that line.

Reply:
Honestly, Deprecating string methods is stupid, as it serves a purpose to send a simple message.

One could argue the legacy codes support of this message are 'deprecated' and that the method isn't, but it's also possible to permanently support that system on the API, which I'm supportive of, as components are stupid for simple messaging. 

I'm in favor of permanently keeping legacy codes around for simple messaging, and if mojang adds more advanced capabilities than current with components, then the legacy system just wouldn't expose it and people can move to comps then.

But even a sendMessage(ChatColor color, String msg); that doesn't use legacy codes would be useful to declare simple messages, but I see no harm in keeping legacy codes on sendMessage.

If md5 removed it from Bukkit API, I would add it back. It costs nothing to maintain.